src/main/react-front-end/src/views/JobHistory/JobHistoryComponent.js

```diff
@@ -0,0 +1,273 @@
+		const {
+			page: prevPage,
+			rowsPerPage: prevRowsPerPage,
+			orderBy: prevOrderBy,
+			order: prevOrder,
+			response: prevResponse,
+			loading: prevLoading
+		} = prevState
+		const {loading, response, page, rowsPerPage, orderBy, order} = this.state
+		if ((!prevLoading && loading !== prevLoading) || response.length !== prevResponse.length ||
+			page !== prevPage || rowsPerPage !== prevRowsPerPage || orderBy !== prevOrderBy ||
+			order !== prevOrder) {
+			this.queueFunc()
+		}
+	}
+
+	// Function that is used to update the table on the page. 
+	// getJobUpdatesForUser takes an array of jobIds and return information about that job
+	update() {
+		const statusSet = new Set(["STARTED", "STARTING", "STOPPED", "STOPPING", "UNKNOWN"])
+		const {responsesToDisplay} = this.state
+		let jobIds = []
+		responsesToDisplay.forEach(job => {
+			if (statusSet.has(job.status) && !jobIds.includes(job.id)) {
+				jobIds.push(job.id)
+			}
+		})
+	}
+
+	paginateResults(results, page, limit) {
+		let offset = page * limit
+		return results.slice(offset, offset + limit)
+	}
+
+	queueFuncSuccess = (resp) => {
+		// const { page, rowsPerPage } = this.state
+		//success
+		//let responsesToDisplay = this.paginateResults(resp.jobs, page, rowsPerPage);
+		//commented to fix second page render issue as it slices all jobs and returns null object
+		
+		// Temporary boolean to check for duplicates changes on backend required
+		let itemsInList = []
+		let dupRemoval = []
+		resp.content.forEach(item => {
+			if (!itemsInList.includes(item.id)) {
+				dupRemoval.push(item)
+				itemsInList.push(item.id)
+			}
+		})
+		console.log(dupRemoval)
+
+		this.setState({
+			response: resp.content,
+			responsesToDisplay: dupRemoval,
+			totalCount: itemsInList.length,
+			loading: false
+		});
+		this.update();
+	}
+
+	queueFuncFail(resp) {
+		//failed
+		console.log(resp)
+		console.log('Error in queue request to API layer');
+	}
+
+	queueFunc(isHistory = false) {
+		const {page, rowsPerPage, orderBy, order} = this.state
+		getJobsForUser(
+			page,
+			rowsPerPage,
+			orderBy,
+			order,
+			this.queueFuncSuccess,
+			this.queueFuncFail
+		);
+	}
+
+	infoButtonOnClick(owner, jobID) {
+		const {selectedRowId} = this.state
+		let identifier = `${owner}-${jobID}`
+		if (selectedRowId && selectedRowId === identifier) {
+			this.setState({selectedRowId: null})
+		} else {
+			this.setState({selectedRowId: identifier})
+		}
+	}
+
+	cancelButtonOnClick(jobID) {
+		cancelJob(jobID, () => {
+			//success
+			this.queueFunc();
+		}, (resp) => {
+			//failed
+			console.log('Error in cancel request to API layer');
+		});
+	}
+
+	restartButtonOnClick(jobID) {
+		restartJob(jobID, () => {
+			//success
+			this.queueFunc()
+		}, () => {
+			//failed
+			var msg = 'Restart job failed since either or both credentials of the job do not exist'
+			console.log(msg)
+			eventEmitter.emit("errorOccured", msg)
+		});
+	}
+
+	deleteButtonOnClick(jobID) {
```

Comment:
For now lets keep this button disabled. I will add in the API calls for it soon.

Reply:
commented it out